RedBeanPHP/Repository.php

```diff
@@ -583,18 +583,43 @@ public function batch( $type, $ids )
 	 */
 	public function convertToBeans( $type, $rows, $mask = NULL )
 	{
-		$masklen = 0;
-		if ( $mask !== NULL ) $masklen = mb_strlen( $mask );
+		$masktype = gettype( $mask );
+		switch ( $masktype ) {
+			case 'string':
+				break;
+			case 'array':
+				$maskflip = array();
+				foreach ( $mask as $m ) {
+					if ( !is_string( $m ) ) {
+						$mask = NULL;
+						$masktype = 'NULL';
+						break 2;
```

Comment:
LOL. I did not know that break X existed ;-)

Reply:
There's continue X as well !